### PR TITLE
Introduce the extension library for Moq to enable convenient experience for mocking

### LINKED ISSUE #3548: [Mocking] Introduce a handy library for our extension method mocking for framework `Moq`
**Describe the issue or request**

The main feature is tracking in https://github.com/Azure/autorest.csharp/issues/3493
This provides a general experience of mocking extension methods in our libraries, but it is not quite easy to use. Therefore we could intorduce a new library to extend the `Moq` framework which is popular.

**Describe your ideas for solutions**

Prototypes are developing and discussing

### PR DESCRIPTION
**Pull request by ArcturusZhang**
Fixes https://github.com/Azure/autorest.csharp/issues/3548

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/resourcemanager/Microsoft.Azure.ResourceManager.Moq/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Newtonsoft.Json support for Azure Core shared client library for .NET
+
+The [Azure.Core package][azure_core_package] contains types shared by the latest Azure SDK client libraries. This `Newtonsoft.Json` compatibility library:
```
Comment (scottaddie) [inline sdk/resourcemanager/Microsoft.Azure.ResourceManager.Moq/README.md:3]: Update the content in this file for the Moq extension library.

File: eng/Packages.Data.props [hunk 2]
```diff
@@ -169,6 +169,11 @@
     <PackageReference Update="Microsoft.Extensions.Options" Version="2.1.0" />
   </ItemGroup>
 
+  <!-- Packages intended for the Moq Extension library only-->
```
Comment (m-nash) [inline eng/Packages.Data.props:172]: we should not have 2 entries for Moq.  We should move Moq into its own conditional entry which checks for `$(MSBuildProjectName) == 'Microsoft.Azure.ResourceManager.Moq'` OR `('$(IsTestProject)' == 'true') OR ('$(IsTestSupportProject)' == 'true') OR ('$(IsPerfProject)' == 'true') OR ('$(IsStressProject)' == 'true') OR ('$(IsSamplesProject)' == 'true')`

We should probably create a property for that really long condition and give it a semantic name like `IsTestReleatedProject`

Comment (jsquire) [reply eng/Packages.Data.props:172]: I'd prefer that we just mark `Microsoft.Azure.ResourceManager.Moq` as a test project than give it some special logic.   After all, it is a package intended to support testing.

Comment (ArcturusZhang) [reply eng/Packages.Data.props:172]: Good idea, I will take a look

File: sdk/resourcemanager/Microsoft.Azure.ResourceManager.Moq/README.md (hunk 1 above)
Comment (ArcturusZhang) [reply sdk/resourcemanager/Microsoft.Azure.ResourceManager.Moq/README.md:3]: I will do that later when I finish the code

**Pull request closed without merging**